client-v2/src/bundles/__tests__/frontsUIBundle.spec.ts

```diff
@@ -288,6 +289,129 @@ describe('frontsUIBundle', () => {
         ).toEqual([0, 1, 2, 3]);
       });
     });
+    describe('selectOpenFrontsCollectionsAndArticles', () => {
+      it('should return just fronts if no collections are open', () => {
+        const openEntities = selectOpenFrontsCollectionsAndArticles(
+          initialStateForOpenFronts
+        );
+        expect(openEntities).toEqual([
+          { collections: [], frontId: 'editorialFront' },
+          { collections: [], frontId: 'editorialFront2' }
+        ]);
+      });
+      it('should give an array of fronts, with nested collections and articles, when those fronts and collections are open -- single collection', () => {
+        const state = set(
+          ['editor', 'collectionIds'],
+          ['collection1'],
+          initialStateForOpenFronts
+        );
+        const openEntities = selectOpenFrontsCollectionsAndArticles(state);
+        expect(openEntities.length).toEqual(2); // Two fronts
+        expect(openEntities[0].collections.length).toEqual(1); // First front has one collection
+        expect(openEntities[0].collections[0].articleIds).toEqual([
+          'articleFragment1',
+          'articleFragment2',
+          'articleFragment3'
+        ]); // First collection has three articles
+        expect(openEntities[1].collections.length).toEqual(0); // Second front has no collections
+      });
+      it('should give an array of fronts, with nested collections and articles, when those fronts and collections are open -- multiple collections', () => {
+        const state = set(
+          ['editor', 'collectionIds'],
+          ['collection1', 'collection6'],
+          initialStateForOpenFronts
+        );
+        const openEntities = selectOpenFrontsCollectionsAndArticles(state);
+        expect(openEntities.length).toEqual(2); // Two fronts
+        expect(openEntities[0].collections.length).toEqual(1); // First front has one collection
```

Comment:
Apologies for the noise here -- I've moved the selector tests in this file to keep them in one place. The relevant test is for `selectDoesCollectionHaveOpenForms`.